Jamulus.pro

```diff
@@ -1048,10 +1103,40 @@ contains(CONFIG, "opus_shared_lib") {
     DISTFILES += $$DISTFILES_OPUS
 }
 
+
+CONFIG(portaudio) {
+    DEFINES += USE_PORTAUDIO
+    HEADERS += src/portaudio-sound.h
+    SOURCES += src/portaudio-sound.cpp
+    mingw* {
+        CONFIG(portaudio_shared_lib) {
+            LIBS += -lportaudio
+        } else {
+            DEFINES += PA_USE_ASIO=1
+            INCLUDEPATH += $$INCLUDEPATH_PORTAUDIO
+            HEADERS += $$HEADERS_PORTAUDIO
+            portaudiocxx.variable_out = OBJECTS
+            portaudiocc.variable_out = OBJECTS
+            QMAKE_EXTRA_COMPILERS += portaudiocc portaudiocxx
+            DISTFILES += $$DISTFILES_PORTAUDIO
+        }
+        LIBS += -lwinmm \
+            -lole32 \
+            -luuid \
+            -lsetupapi
+    } else {
+        error( "portaudio only tested on mingw for now" )
+    }
+}
+
 # disable version check if requested
 contains(CONFIG, "disable_version_check") {
     message(The version check is disabled.)
     DEFINES += DISABLE_VERSION_CHECK
 }
 
 ANDROID_ABIS = armeabi-v7a arm64-v8a x86 x86_64
+
+CONFIG += silent
+QMAKE_CXXFLAGS += -Wno-deprecated-declarations
```

Comment:
Hmm... Do we have deprecated warnings in the code? Where do they come from? From the portaudio library or the Jamulus code?

Reply:
I was getting a bunch of deprecation warnings from Qt, I guess you might be compiling with an older Qt version? I have the this and the `CONFIG += silent` thing in a separate commit so it's easy to drop and/or add it separately. Mainly these are there to keep myself sane.

    ../src/util.h:1303:109: warning: 'QTextStream& QTextStreamFunctions::endl(QTextStream&)' is deprecated: Use Qt::endl [-Wdeprecated-declarations]
     1303 |                             streamFile << i << " " << static_cast<double> ( vElapsedTimes[i] ) / 1000000 << endl;
          |                                                                                                             ^~~~
    In file included from C:/msys2/mingw64/include/QtCore/qdebug.h:49,
                     from C:/msys2/mingw64/include/QtNetwork/qabstractsocket.h:47,
                     from C:/msys2/mingw64/include/QtNetwork/qhostaddress.h:48,
                     from C:/msys2/mingw64/include/QtNetwork/QHostAddress:1,
                     from ../src/client.h:27,
                     from ../src/client.cpp:25:
    C:/msys2/mingw64/include/QtCore/qtextstream.h:293:75: note: declared here
      293 | Q_CORE_EXPORT QT_DEPRECATED_VERSION_X(5, 15, "Use Qt::endl") QTextStream &endl(QTextStream &s);
          |                                                                           ^~~~

Reply:
Sounds like your branch could use a rebase against master? :) The latter includes #830 which probably fixes the same issue (although with a slightly different approach).

(Glad to see the work on Portaudio, btw! :))

Reply:
> The latter includes #830 which probably fixes the same issue (although with a slightly different approach).

Ah, yes. Looks like a better approach too since it's targeted only at Qt.